### PR TITLE
add  "fbask_jan16_sticky" realization to livestock module

### PR DESCRIPTION
**Pull request by DavidhoPIK**
## :bird: Description of this PR :bird:

This PR adds the ```fbask_jan16_sticky``` realization to the livestock module. It is a "sticky" version of the ```fbask_jan16``` realization, that uses the (simplified) methodology of the ```sticky_feb18``` factor cost realization for it's factor costs calculation. 

The documentation describes the difference to ```fbask_jan16``` as follows: 
```
In this realization the capital share of livestock production cost does not have 
to be paid every timestep, but is fulfilled by having a corresponding capital stock,
which mostly persists across timesteps. (See `s70_depreciation_rate`)
This means that the production becomes cheaper in regions with
higher capital shares and existing corresponding capital stocks,
which makes the livestock production "stick" to some degree to existing levels. 
Another effect is that production increases are more likely to occur in regions
of more capital intensive livestock systems with already established stocks.

The realization is based on the `sticky_feb18` factor cost realization, but
also differs from it in some ways. At the creation time the main differences were: 
1. capital stocks are on the regional level not on cluster level
2. no differentiation of mobile and immobile capital - all capital stocks are immobile
```

## :wrench: Checklist for PR creator :wrench:

- [x] Label pull request [from the label list](https://github.com/magpiemodel/magpie/labels).
  - **Low risk**: Simple bugfixes (missing files, updated documentation, typos) or changes  in start or output scripts
  - **Medium risk**: Uncritical changes in the model core (e.g. moderate modifications in non-default realizations)
  - **High risk**: Critical changes in model core or default settings (e.g. changing a model default or adjusting a core mechanic in the model)

- [x] Self-review own code
  - No hard coded numbers and cluster/country/region names.
  - The new code doesn't contain declared but unused parameters or variables.
  - [`magpie4`](https://github.com/pik-piam/magpie4) R library has been updated accordingly and backwards compatible where necessary.
  - `scenario_config.csv` has been updated accordingly (important if `default.cfg` has been updated)

- [x] Document changes 
  - Add changes to `CHANGELOG.md`
  - Where relevant, put In-code documentation comments
  - Properly address updates in interfaces in the module documentations
  - run [`goxygen::goxygen()`](https://github.com/pik-piam/goxygen) and verify the modified code is properly documented

- [x] Perform test runs
  - **Low risk**: 
    - Run a compilation check via `Rscript start.R --> "compilation check"`
  - **Medium risk**: 
    - Run test runs via `Rscript start.R --> "test runs"`
    - Check logs for errors/warnings
  - **High risk**:
    - Run test runs via `Rscript start.R --> "test runs"`
    - Check logs for errors/warnings
    - Default run from the PR target branch for comparison
    - Provide relevant comparison plots (land-use, emissions, food prices, land-use intensity,...)

### :chart_with_downwards_trend: Performance changes :chart_with_upwards_trend:
  
  - Current develop branch default (gams execution): 20.0 mins
  - This PR's default :  19.5mins

## :rotating_light: Checklist for reviewer :rotating_light:

- PR is labeled correctly
- Code changes look reasonable
  - No hard coded numbers and cluster/country/region names.
  - No unnecessary increase in module interfaces
  - model behavior/performance is satisfactory.
- Changes are properly documented
  - `CHANGELOG` is updated correctly
  - Updates in interfaces have been properly addressed in the module documentations
  - In-code documentation looks appropriate
- [ ] content review done (at least 1)
- [x] RSE review done (at least 1)

### CODE REVIEW

**Comment by DavidhoPIK** [comment]
The main intended effect of the realization is that it provides more stability to the livestock production, in case of more liberal trade.
The largest differences are visible when using superregions, but such tests where not possible since the h16s5 preprocessing had problems. However to some extend the stabilizing effect can also be seen with ```cfg$gms$c21_trade_liberalization <- "l908080r807070"```. Here are plots of relevant variables which are filtered by the condition of actually showing significant differences:

image

image

image

image

image

**Comment by DavidhoPIK** [comment]
Here are some more plots of "standard" variables for completeness: 
image

image

image

**Comment by DavidhoPIK** [comment]
The only warnings/errors from the GAMS model run I got where: 
```
**** ERRORS/WARNINGS IN VARIABLE vm_land(MEA_93,primforest)
     1 error(s): Variable considered fixed at the initial value projected on the bounds.
**** ERRORS/WARNINGS IN VARIABLE vm_land(EUR_43,primforest)
     1 error(s): Variable considered fixed at the initial value projected on the bounds.
```
I think these errors are unrelated to the new realization. They also do not stop the execution of the GAMS code.

File: modules/70_livestock/fbask_jan16_sticky/realization.gms [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+*' to be paid every timestep, but is fulfilled by having a corresponding capital stock,
+*' which mostly persists across timesteps. (See `s70_depreciation_rate`)
+*' This means that the production becomes cheaper in regions with
+*' higher capital shares and existing corresponding capital stocks,
+*' which makes the livestock production "stick" to some degree to existing levels. 
+*' Another effect is that production increases are more likely to occur in regions
+*' of more capital intensive livestock systems with already established stocks.
+*'
+*' The realization is based on the `sticky_feb18` factor cost realization, but
+*' also differs from it in some ways. At the creation time the main differences where:
```
Comment (tscheypidi) [inline modules/70_livestock/fbask_jan16_sticky/realization.gms:91]:
```suggestion
*' also differs from it in some ways. At the creation time the main differences were: 
```

**Review by tscheypidi — APPROVED** [review]
looks good to me. Found just one typo.

**Comment by DavidhoPIK** [comment]
I decided to do some another evaluation of the new realization with more more extreme settings, to show in which direction the effects go. The following plots are with the settings: 
```
cfg$gms$c21_trade_liberalization <- "free2000"

cfg$gms$s70_multiplicator_capital_need <- 3
cfg$gms$s70_depreciation_rate <- 0.02
```
So we have a higher capital need, a lower depreciation of stocks and a very free trade. Of course these settings are unrealistic but they serve to show how the main effect of using the realization: It then takes time to build up capital stocks and production is not so easily shifted away from regions when capital stocks are already build up. 

image

image

image

image

image

image

image

File: modules/70_livestock/fbask_jan16_sticky/presolve.gms
Comment (emolinab) [inline modules/70_livestock/fbask_jan16_sticky/presolve.gms:1]: Why is there a s70_multiplicator_capital_need?

**Review by emolinab — COMMENTED** [review]
It looks good and very cool to me! I am missing the difference in run time with the implementation activated, and it is not so clear why a s70_multiplicator_capital_need is needed.

**Comment by DavidhoPIK** [comment]
Thank you for the review and comment! There was no significant difference in the runtime 19,5 min with the realization 20 min without (only checked for the first test run i did).

File: modules/70_livestock/fbask_jan16_sticky/presolve.gms
Comment (DavidhoPIK) [reply modules/70_livestock/fbask_jan16_sticky/presolve.gms:1]: I added the  ````s70_multiplicator_capital_need```` multiplikator to reflect uncertainty that we may have in the capital need. 

So the default capital need is such that multiplied with the sum of interest rate and depreciation rate it equals the capital factor requirements. This means it strongly depends on the depreciation rate which is also configurable and a global estimate. 

In general my guess would be that it is difficult to estimate the capital need from the VoP. (Correct me if i'm wrong!)

**Review by emolinab — APPROVED** [review]

**Pull request merged**